test/steps/v2_steps.py

```diff
@@ -711,12 +883,228 @@ def expect_path(context, path):
+        with open(dir_path + "/test-harness/features/resources/" + approval_program, "rb") as f:
+            approval_program = bytearray(f.read())
+    if clear_program == "none":
+        clear_program = None
+    elif clear_program:
+        with open(dir_path + "/test-harness/features/resources/" + clear_program, "rb") as f:
+            clear_program = bytearray(f.read())
+    if app_args == "none":
+        app_args = None
+    elif app_args:
+        app_args = split_and_process_app_args(app_args)
+    if foreign_apps == "none":
+        foreign_apps = None
+    elif foreign_apps:
+        foreign_apps = [int(app) for app in foreign_apps.split(",")]
+    if app_accounts == "none":
+        app_accounts = None
+    elif app_accounts:
+        app_accounts = [account_pubkey for account_pubkey in app_accounts.split(",")]
+    if genesis_hash == "none":
+        genesis_hash = None
+    if int(local_ints) == 0 and int(local_bytes) == 0:
+        local_schema = None
+    else:
+        local_schema = transaction.StateSchema(num_uints=int(local_ints), num_byte_slices=int(local_bytes))
+    if int(global_ints) == 0 and int(global_bytes) == 0:
+        global_schema = None
+    else:
+        global_schema = transaction.StateSchema(num_uints=int(global_ints), num_byte_slices=int(global_bytes))
+    sp = transaction.SuggestedParams(int(fee), int(first_valid), int(last_valid), genesis_hash, flat_fee=True)
+    context.transaction = transaction.ApplicationCallTxn(sender=sender, sp=sp, index=int(application_id),
+                                                         on_complete=operation, local_schema=local_schema,
+                                                         global_schema=global_schema,
+                                                         approval_program=approval_program, clear_program=clear_program,
+                                                         app_args=app_args, accounts=app_accounts,
+                                                         foreign_apps=foreign_apps,
+                                                         note=None, lease=None, rekey_to=None)
+
+
+@when('sign the transaction')
+def sign_transaction_with_signing_account(context):
+    private_key = mnemonic.to_private_key(context.signing_mnemonic)
+    context.signed_transaction = context.transaction.sign(private_key)
+
+
+@then('the base64 encoded signed transaction should equal "{golden}"')
+def compare_to_base64_golden(context, golden):
+    actual_base64 = encoding.msgpack_encode(context.signed_transaction)
+    assert (golden == actual_base64)
+
+
+@given('an algod v2 client connected to "{host}" port {port} with token "{token}"')
+def step_impl(context, host, port, token):
+    algod_address = "http://" + str(host) + ":" + str(port)
+    context.app_acl = algod.AlgodClient(token, algod_address)
+    # also need a v1 client for things like transaction_info
+    context.acl = legacyclient.AlgodClient(token, algod_address)
```

Comment:
context.acl is already initialized by the "Background", no need to re-initialize it here